src/inmemory_cache.rs

```diff
@@ -0,0 +1,136 @@
+use std::{sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+
+use crate::{
+    block::Block, db_state::SsTableId, filter::BloomFilter, flatbuffer_types::SsTableIndexOwned,
+};
+
+#[derive(Clone)]
+pub(crate) enum CachedBlock {
+    Block(Arc<Block>),
+    #[allow(dead_code)]
+    Index(Arc<SsTableIndexOwned>),
+    #[allow(dead_code)]
+    Filter(Arc<BloomFilter>),
+}
+
+#[derive(Clone, Copy)]
+pub enum CacheType {
+    Moka,
+}
+
+#[derive(Clone, Copy)]
+pub struct InMemoryCacheOptions {
+    pub max_capacity: u64,
+    pub cached_block_size: u32,
+    pub time_to_live: Option<Duration>,
+    pub time_to_idle: Option<Duration>,
+    pub cache_type: CacheType,
+}
+
+impl Default for InMemoryCacheOptions {
+    fn default() -> Self {
+        Self {
+            max_capacity: 64 * 1024 * 1024, // 64MB default max capacity
+            cached_block_size: 32,          // 32 bytes default
+            time_to_live: None,
+            time_to_idle: None,
+            cache_type: CacheType::Moka, // Default to Moka cache
+        }
+    }
+}
+
+#[async_trait]
+pub(crate) trait InMemoryCache: Send + Sync + 'static {
```

Comment:
@pragmaticanon @criccomini @vigneshc @flaneur2020 what do you guys think about renaming this interface to `BlockCache`. There's nothing that requires the implementation to be a purely in-memory cache. It's just defining an interface for a cache that caches different kinds of db blocks.

Reply:
😂 I suggested renaming it from BlockCache because it can cache indexes and filters too. `Cache` seemed confusing since we're about to have the disk cache separately. 🤔

Reply:
Ah. I was thinking `BlockCache` because I was thinking of indexes/filters as just different types of blocks in ssts (though we're really overloading the term "block"). Maybe `DbCache` to specify it's caching db things rather than object store things?

Reply:
> I was thinking BlockCache because I was thinking of indexes/filters as just different types of blocks

Haha, yea I considered that. But we literally have `block.rs` with `Block`'s. Then I was like .. "Maybe Block should be Page" and I said, "No Chris. Put down the paint, step away from the bike shed."

I do like DbCache. That sounds right-ish to me. I'll open an issue.

Reply:
https://github.com/slatedb/slatedb/issues/191